avocado/utils/configure_network.py

```diff
@@ -30,7 +30,7 @@ class NWException(Exception):
     """
 
 
-def set_ip(ipaddr, netmask, interface):
+def set_ip(ipaddr, netmask, interface, interface_type):
     """
```

Comment:
My suggestion is to use some default value for interface_type. You can use 'None' and make it default to 'Ethernet' on the condition to write the file, or you can use interface_type='Ethernet' and write "TYPE="+interface_type+" \n," avoiding the if/elif/elif/... statements as interface_type possibilities grow.

Reply:
@willianrampazzo sure.

Reply:
I'm not sure if it is the case for this function, but the proposed code won't help in case I want to set a bridge device with 'TYPE=Bridge'. That is the reason I suggested interface_type as a string, this way the code is not limited by the options available on it.

Reply:
@willianrampazzo the code that I see seems to make what you suggest possible. That is:

```python
set_ip('10.1.1.1', '255.255.255.0', 'eth0')
```

Should result in:

```
...
TYPE=Ethernet
...
```

While:

```python
set_ip('10.1.1.1', '255.255.255.0', 'eth0', 'Bridge')
```

Should result in:

```
...
TYPE=Bridge
...
```

But, I'd suggest a different style in the code:

```python
if interface_type is None:
    interface_type = 'Ethernet'
network_conf.write("TYPE=%s \n" % interface_type)
```

Reply:
That is absolutely true. I misread the code. Where is written `network_conf.write("TYPE=%s \n" % interface_type)`, I read `network_conf.write("TYPE=InfiniBand \n")`. My bad.

Besides the coding style mentioned, code looks good to me.